streams-contrib/streams-persist-mongo/src/main/java/org/apache/streams/mongo/MongoPersistReader.java

```diff
@@ -133,7 +132,18 @@ public void cleanUp() {
 
     protected StreamsDatum prepareDatum(DBObject dbObject) {
 
-        StreamsDatum datum = new StreamsDatum(dbObject.toString());
+        ObjectNode objectNode;
+        String id;
+
+        try {
+            objectNode = mapper.readValue(dbObject.toString(), ObjectNode.class);
+            id = objectNode.get("_id").get("$oid").asText();
```

Comment:
Based on the internal structure, I'm guessing that you're getting an ObjectId object back.  It might be best to use that instead of using ObjectNode.  My fear is that this is brittle and MongoDb could change its internals and $oid may not be available anymore.